### PR TITLE
Auto-Generate Docs for CLI

### PR DESCRIPTION
**Pull request by joshuajerin**

### CODE REVIEW

File: .github/actions/generate-docs/action.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      shell: bash
+
+    - name: Commit and push documentation to website repository
+      run: |
+        cd website-repo
+        git config user.name "coredb-service-user"
+        git config user.email "admin@github.com"
+        git add src/content/docs/development/cli/command-reference.md
+        git checkout -b ${{ inputs.website_branch }}
+        git commit -m "Update command reference documentation" && git push origin ${{ inputs.website_branch }} || echo "No change in docs!"
```
Comment (sjmiller609) [inline .github/actions/generate-docs/action.yml:74]: Can you please include the short git commit hash of the tembo CLI commit in the commit message, like this.

This part https://github.com/tembo-io/tembo/blob/c6e96c80d27f5dc119260c671061e6f915de36ed/.github/workflows/deploy.yml#L23

should run on the code checkout of the CLI

Then you have the short SHA in an "output" of a step. Then you can include it like this

```
git commit -m "Update command reference documentation: ${{ steps.versions.outputs.SHORT_SHA }}" && git push origin ${{ inputs.website_branch }} || echo "No change in docs!" 
```

File: .github/workflows/tembo_cli.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      - 'tembo-cli/**'
+    branches:
+      - main
+  pull_request:
+    paths:
+      - 'tembo-cli/**'
+    branches:
+      - main
+      - develop
+      - feature/*
```
Comment (sjmiller609) [inline .github/workflows/tembo_cli.yml:15]: Delete this before merge

File: .github/workflows/tembo_cli.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+          run: |
+            set -xe
+            sudo apt-get update
+
+        - name: Generate and Deploy Docs
+          uses: ./.github/actions/generate-docs
+          with:
+            tembo_repository: 'tembo-io/tembo'
+            ssh_key: ${{ secrets.SERVICE_USER_GITHUB_SSH_KEY }}
+            tembo_branch: 'plat-410'
```
Comment (sjmiller609) [inline .github/workflows/tembo_cli.yml:32]: update this before merge

File: tembo-cli/Cargo.toml [hunk 4]
```diff
@@ -76,6 +76,7 @@ tembo-stacks = "0.4.1"
 itertools = "0.12.1"
 random-string = "1.1.0"
 test-case = "=2.0.0-rc2"
+clap-markdown = { git = "https://github.com/joshuajerin/clap-markdown.git", branch = "0.1.4" }
```
Comment (sjmiller609) [inline tembo-cli/Cargo.toml:79]: Can you move this to a tembo repository before merge

File: tembo-cli/command-reference.md [hunk 5]
```diff
@@ -0,0 +1,182 @@
+---
```
Comment (sjmiller609) [inline tembo-cli/command-reference.md:1]: Can you delete this file from the repo, please

**Review by sjmiller609 — CHANGES_REQUESTED** [review]

File: tembo-cli/Cargo.toml (hunk 4 above)
Comment (joshuajerin) [reply tembo-cli/Cargo.toml:79]: done

**Review by sjmiller609 — APPROVED** [review]

File: .github/actions/generate-docs/action.yml [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+  website_branch:
+    description: 'Branch to update with generated documentation.'
+    required: true
+
+on:
+  push:
+    paths:
+      - 'tembo-cli/**'
+    branches:
+      - main
```
Comment (sjmiller609) [inline .github/actions/generate-docs/action.yml:25]: This should be removed

File: .github/workflows/tembo_cli.yml [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+      - 'tembo-cli/**'
+    branches:
+      - main
+  pull_request:
+    paths:
+      - 'tembo-cli/**'
+    branches:
+      - main
+      - develop
+      - feature/*
```
Comment (sjmiller609) [inline .github/workflows/tembo_cli.yml:15]: Please delete this block

**Review by sjmiller609 — APPROVED** [review]

**Pull request merged**